src/compiler/crystal/semantic/top_level_visitor.cr

```diff
@@ -793,6 +793,10 @@ class Crystal::TopLevelVisitor < Crystal::SemanticVisitor
     const = Const.new(@program, scope, name, value)
     const.private = true if target.visibility.private?
 
+    process_annotations(annotations) do |annotation_type, ann|
+      const.add_annotation(annotation_type, ann) if annotation_type == @program.deprecated_annotation
```

Comment:
Please add a code comment why only `Deprecated` is supported here (maybe just a link to this PR?).